internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,266 @@
+			Operations:    originalTx.Operations(),
+			BaseFee:       int64(t.BaseFee),
+			Preconditions: txnbuild.Preconditions{
+				TimeBounds: txnbuild.NewTimeout(10),
+			},
+			IncrementSequenceNum: true,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building transaction: %w", err)
+	}
+	tx, err = t.ChannelAccountSignatureClient.SignStellarTransaction(ctx, tx, channelAccountPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("signing transaction with channel account: %w", err)
+	}
+	// Wrap the transaction in a fee bump tx, signed by the distribution account
+	distributionAccountPublicKey, err := t.DistributionAccountSignatureClient.GetAccountPublicKey(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting distribution account public key: %w", err)
+	}
+
+	feeBumpTx, err := txnbuild.NewFeeBumpTransaction(
+		txnbuild.FeeBumpTransactionParams{
+			Inner:      tx,
+			FeeAccount: distributionAccountPublicKey,
+			BaseFee:    int64(t.BaseFee),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building fee-bump transaction %w", err)
+	}
+
+	feeBumpTx, err = t.DistributionAccountSignatureClient.SignStellarFeeBumpTransaction(ctx, feeBumpTx)
+	if err != nil {
+		return nil, fmt.Errorf("signing the fee bump transaction with distribution account: %w", err)
+	}
+	return feeBumpTx, nil
+}
+
+func (t *transactionService) parseErrorResultXDR(errorResultXdr string) (tss.RPCTXCode, error) {
+
+	//errorResult := xdr.TransactionResult{}
+	unMarshallErr := "unable to unmarshal errorResultXdr: %s"
+	//err := errorResult.UnmarshalBinary([]byte(errorResultXdr))
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(errorResultXdr)
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf(unMarshallErr, errorResultXdr)
+	}
+	dec := xdr3.NewDecoder(strings.NewReader(string(decodedBytes)))
+	var errorResult xdr.TransactionResult
+	_, err = dec.Decode(&errorResult)
+
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf(unMarshallErr, errorResultXdr)
+	}
+	return tss.RPCTXCode{
+		TxResultCode: errorResult.Result.Code,
+	}, nil
+}
+
+func (t *transactionService) SendTransaction(transactionXdr string) (tss.RPCSendTxResponse, error) {
+	rpcResponse, err := t.sendRPCRequest("sendTransaction", map[string]string{"transaction": transactionXdr})
+	sendTxResponse := tss.RPCSendTxResponse{}
+	sendTxResponse.TransactionXDR = transactionXdr
+	if err != nil {
+		sendTxResponse.Code.OtherCodes = tss.RPCFailCode
+		return sendTxResponse, fmt.Errorf(err.Error())
+	}
+
+	if result, ok := rpcResponse["result"].(map[string]interface{}); ok {
+		if val, exists := result["status"].(string); exists {
+			sendTxResponse.Status = tss.RPCTXStatus(val)
+		}
+		if val, exists := result["errorResultXdr"].(string); exists {
+			sendTxResponse.Code, err = t.parseErrorResultXDR(val)
+		}
+		if hash, exists := result["hash"].(string); exists {
+			sendTxResponse.TransactionHash = hash
+		}
+	} else {
```

Comment:
In `sendRPCRequest` we could map the RPC response to a `RPCResponse` struct, wdyt?
So that we don't have to work with these maps everywhere. Seems like the response format is somewhat standard with a `result` field, `status`, etc.

Reply:
Take a look at what I did. I have this method now return a newly created RPCResponse struct that all possilble rpc fields get unmarshaled into. However, the sendTransaction and getTransaction methods will do some further processing (calling parseErrorResultXDR for example) and return respective structs. 

I could have done ALL of that inside the sendRPCRequest method, essentially switching on method type, but I did not want to bloat that function. I wanted sendTransaction related validation to be done in the SendTransaction method, getTransaction related validation in the GetTransaction method